src/extension.ts

```diff
@@ -63,3 +86,76 @@ const getComponentName = (): Thenable<string | undefined> =>
         : "Please enter a valid component name";
     },
   });
+
+/**
+ * Check availability of Angular CLI on the command line
+ * @param directory Directory which will use the Angular CLI
+ */
+const checkAngularCli = (directory: string) => {
+  try {
+    execSync("ng --version", { cwd: directory });
+  } catch (error) {
+    throw Error("Could not find Angular CLI, Error: " + error);
+  }
+};
+
+/**
+ * Get the name of the directory of the current file
+ * @returns Name of the directory
+ */
+const getDirectoryName = () => {
```

Comment:
(Optional) I think it would be better to return `undefined` instead of throwing an error. Then you can simplify the caller code as well to:

```ts
const componentDirectory = getDirectoryName();
if(componentDirecoty === undefined){
  vscode.window.showErrorMessage(
    `Could not find directory to generate component ${componentName}`
  );
  return;
}
```

Reply:
Yes I agree with you, returning `undefined` seems to be better.